File: manashelper-bot/src/main/java/kg/manasuniversity/usbtypec/manashelper/service/DepartmentService.java
package kg.manasuniversity.usbtypec.manashelper.service;

import kg.manasuniversity.usbtypec.manashelper.entity.Department;
import kg.manasuniversity.usbtypec.manashelper.model.DepartmentSummary;
import kg.manasuniversity.usbtypec.manashelper.repository.DepartmentRepository;
import lombok.RequiredArgsConstructor;
import org.springframework.stereotype.Service;

import java.util.List;
import java.util.UUID;

@Service
@RequiredArgsConstructor
public class DepartmentService {
    private final DepartmentRepository departmentRepository;

    public List<DepartmentSummary> getDepartmentsByFaculty(UUID facultyId) {
        return departmentRepository.findAllByFacultyId(facultyId).stream()
            .map(this::toSummary)
            .toList();
    }

    private DepartmentSummary toSummary(Department department) {
        return new DepartmentSummary(department.getId(), department.getName());
    }
}

File: src/manashelper/services/department_service.py
import uuid
from dataclasses import dataclass

from manashelper.repositories.department_repository import DepartmentRepository


@dataclass(frozen=True, slots=True)
class DepartmentSummary:
    id: uuid.UUID
    name: str


class DepartmentService:
    def __init__(self, department_repository: DepartmentRepository) -> None:
        self._department_repository = department_repository

    async def get_departments_by_faculty(self, faculty_id: uuid.UUID) -> list[DepartmentSummary]:
        departments = await self._department_repository.get_all_by_faculty_id(faculty_id)
        return [DepartmentSummary(id=department.id, name=department.name) for department in departments]
